environments/dev-env/install_req.sh

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env bash
+# Copyright 2023 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+get_os_type() {
+  case "$OSTYPE" in
+      darwin*)  echo "darwin" ;;
+      linux*)   echo "linux" ;;
+      *)        echo "unknown: $OSTYPE" && exit 1 ;;
+  esac
+}
+
+os_type=$(get_os_type)
+
+# Install golang for x86_64
+if [ $os_type == "linux" ]; then
+    if !(command -v go) then
+        sudo apt update
+        sudo apt upgrade -y
+        curl -O https://dl.google.com/go/go1.19.2.linux-amd64.tar.gz
+        tar -xvf go1.19.2.linux-amd64.tar.gz
+        sudo mv go  /usr/bin/
+        sudo echo 'export GOROOT=/usr/bin/go' >> ~/.bashrc 
+        sudo echo 'export PATH=$PATH:$GOROOT/bin' >> ~/.bashrc 
+        sleep 5
+        exec bash
+    else
+        echo "Go is already installed"
+        echo "***********************"
+    fi
+elif [ $os_type == "darwin" ]; then
+    if !(command -v go) then
+        brew update
+        curl -O https://dl.google.com/go/go1.19.2.darwin-amd64.pkg
+        tar -xvf go1.19.2.darwin-amd64.pkg
+        sudo mv go  /usr/bin/
+        sudo echo 'export GOROOT=/usr/bin/go' >> ~/.bashrc 
+        sudo echo 'export PATH=$PATH:$GOROOT/bin' >> ~/.bashrc 
+        sleep 5
+        exec bash
+    else
+        echo "Go is already installed"
+        echo "***********************"
+    fi
+fi
+
+
+# Install make
+if [ $os_type == "linux" ]; then
+    if !(dpkg -l make) then
+        apt update
+        apt install -y make
+    else
+    echo "Make is already installed"
+    echo "*************************"
+    fi
+elif [ $os_type == "darwin" ]; then
+    if !(command -v make) then
+        brew update
```

Comment:
Is it desirable to explicitly invoke `brew update` in every copy of this pattern?  I find that brew itself will automatically update if that has not been done recently.  Forcing several `brew update`s in quick succession seems to me like a waste of time.